### PR TITLE
Script to add Uperf which is a network benchmark tool to measure

### PR DESCRIPTION
**Pull request by hathyaga**
Network performance

This script will calculate the performance of the network IO adapter
in GB

Signed-off-by: Harsha Thyagaraja <harshkid@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by hathyaga** [comment]
Job log:
[multiport_job_log_PASS.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/1387678/multiport_job_log_PASS.txt)

File: io/net/uperf_test.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        """
+        To check and install dependencies for the test
+        """
+        smm = SoftwareManager()
+        detected_distro = distro.detect()
+        pkgs = ["gcc", "autoconf", "perl", "m4", "git-core"]
+        if detected_distro.name == "Ubuntu":
+            pkgs.extend(["libsctp1", "libsctp-dev", "lksctp-tools"])
+        else:
+            pkgs.extend(["lksctp-tools", "lksctp-tools-devel"])
```
Comment (narasimhan-v) [inline io/net/uperf_test.py]: are the package names same for all other distros ?

File: io/net/uperf_test.py.data/README.txt [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+Requirements:
+-----------------------
+1. Generate sshkey for your test partner to run the test uninterrupted.
+2. Install netifaces using pip. command: pip install netifaces
+3. Install dependant packages for uperf tools to be compiled in the
+Peer machine. 
+For Rhel and Sles distros: lksctp-tools, lksctp-tools-devel
+For Ubuntu: libsctp1, libsctp-dev, lksctp-tools
+
```
Comment (narasimhan-v) [inline io/net/uperf_test.py.data/README.txt]: unwanted empty lines

File: io/net/uperf_test.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+        """
+        if self.uperf_run == '1':
+            tmp = "cd /tmp/uperf-master/src;./uperf -s"
+            cmd = "ssh %s@%s \"%s\"" % (self.peer_user, self.peer_ip, tmp)
+            obj = process.SubProcess(cmd, verbose=False, shell=True)
+            obj.start()
+        speed = int(read_file("/sys/class/net/%s/speed" % self.iface))
+        self.expected_tp = int(self.expected_tp) * speed / 100
+        os.chdir(self.uperf)
+        time.sleep(1)
```
Comment (narasimhan-v) [inline io/net/uperf_test.py]: why sleep here ?

File: io/net/uperf_test.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+        cmd = "h=%s proto=tcp uperf -m throughput.xml -a" % self.peer_ip
+        result = process.run(cmd, shell=True, ignore_status=True)
+        if result.exit_status != 0:
+            self.fail("FAIL: Uperf Run failed")
+        for line in result.stdout.splitlines():
+            if self.peer_ip in line:
+                tmp = int(line.split()[3].split('.')[0])
+                # Converting the throughput calculated in Gb to Mb
+                tput = tmp * 1000
```
Comment (narasimhan-v) [inline io/net/uperf_test.py]: It can be `tput = int(line.split()[3].split('.')[0]) * 1000`

Comment (narasimhan-v) [inline io/net/uperf_test.py]: Are we sure the reported throughput is always in Gb ?

File: io/net/uperf_test.py [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+            cmd = "ssh %s@%s \"%s\"" % (self.peer_user, self.peer_ip, tmp)
+            obj = process.SubProcess(cmd, verbose=False, shell=True)
+            obj.start()
+        speed = int(read_file("/sys/class/net/%s/speed" % self.iface))
+        self.expected_tp = int(self.expected_tp) * speed / 100
+        os.chdir(self.uperf)
+        time.sleep(1)
+        cmd = "h=%s proto=tcp uperf -m throughput.xml -a" % self.peer_ip
+        result = process.run(cmd, shell=True, ignore_status=True)
+        if result.exit_status != 0:
```
Comment (apahim) [inline io/net/uperf_test.py]:
```python
>>> var = 0
>>> bool(var)
False

>>> var = 1
>>> bool(var)
True

>>> var = -1
>>> bool(var)
True
```
So...
```python
        if result.exit_status:
            self.fail(...)

```

File: io/net/uperf_test.py [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+        if 'WARNING' in result.stdout:
+            self.log.warn('Test completed with warning')
+
+    def tearDown(self):
+        """
+        Killing Uperf process in peer machine
+        """
+        msg = "pkill uperf; rm -rf /tmp/uperf-master"
+        cmd = "ssh %s@%s \"%s\"" % (self.peer_user, self.peer_ip, msg)
+        if process.system(cmd, shell=True, ignore_status=True) != 0:
```
Comment (apahim) [inline io/net/uperf_test.py]: ditto

**Comment by hathyaga** [comment]
New PR: https://github.com/avocado-framework-tests/avocado-misc-tests/pull/670

**Pull request closed without merging**